работа над проектом

mchvell opened the pull request:
добавлены:
- тесты
- улучшен стандартный метод из задачи (разбит на 2)
- добавлен отчет

Commit:
работа над проектом

добавлены:
- тесты
- улучшен стандартный метод из задачи (разбит на 2)
- добавлен отчет

AlexKitIt commented on tests/test_auth_courier.py:
```diff
@@ -0,0 +1,55 @@
+
+
+class TestCourierAuth:
+    path = "https://qa-scooter.praktikum-services.ru"
+    method = "/api/v1/courier/login"
```
Нужно исправить: в тестовых классах данные не хранятся, для этого используются data модули

AlexKitIt commented on tests/test_auth_courier.py:
```diff
@@ -0,0 +1,55 @@
+        payload = {"login": user_data[0], "password": user_data[1]}
+
+        response = requests.post(url=self.path+self.method, data=payload)
+        valid_status_code = 200
+        assert response.status_code == valid_status_code
```
Нужно исправить: здесь и далее: одной проверки на статус код недостаточно, необходимо парсить ответ

mchvell replied:
так нужно 2 ассерта? в проекте на pageObject у меня не приняли работу из-за этого.

AlexKitIt replied:
Либо два, либо 2 условия в одном

AlexKitIt commented on tests/test_auth_courier.py:
```diff
@@ -0,0 +1,55 @@
+        payload = {"login": user_data[0], "password": user_data[1]}
+
+        response = requests.post(url=self.path+self.method, data=payload)
+        error_text = '{"code":404,"message":"Учетная запись не найдена"}'
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации

AlexKitIt commented on tests/test_create_couriers.py:
```diff
@@ -0,0 +1,67 @@
+    @allure.description("Вызываем метод /api/v1/courier и проверяем, нельзя создать нового курьера на существующие креды")
+    def test_create_courier_with_existing_data(self):
+        payload = {
+            "login": "wukongTheMonkey",
+            "password": "banana123",
+            "firstName": "StoneMonkey"
+        }
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

AlexKitIt commented on user_generator.py:
```diff
@@ -0,0 +1,62 @@
+def register_new_courier_and_return_login_password():
+    courier_data = generate_random_courier_data()
+
+    response = requests.post('https://qa-scooter.praktikum-services.ru/api/v1/courier', data=courier_data)
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются, а тк же конкатенировать с базовым, так будет проще поддерживать несколько стендов

mchvell replied:
согласен, но это стандартный метод практикума, который изначально был еще более странным и перегруженным. я его сделал чуть умнее, но менять или тюнить до идеала не стал

Commit:
Правки после код-ревью

AlexKitIt commented on data/data_create_courier.py:
```diff
@@ -0,0 +1,13 @@
+class CreateCourierData:
+    path = "https://qa-scooter.praktikum-services.ru"
```
Копипаста не очень хорошо выглядит

mchvell replied:
~~поскольку нет тега исправить или можно лучше – не исправлено. хочется чуть больше вводных по тому, что править~~

~~удалил класс для работы с GET// Order и почистил чуть логику. ~~

~~В целом про копипаст не совсем понятно, что бы ты предложил исправить – вынести в базовый класс одну переменную с урлом? Я понимаю, что возможно это правило хорошего тона, но делать один класс с одной переменной в данном случае ООП ради ООП имхо~~

Сделал класс с базовым урлом в правках v2.3

AlexKitIt commented on tests/test_auth_courier.py:
```diff
@@ -0,0 +1,52 @@
+
+        response = requests.post(url=auth_data.url, data=payload)
+        valid_status_code = 200
+        assert response.status_code == valid_status_code
```
Нужно исправить: повторный коммент: здесь и далее: проверки тела ответа не добавлены

Commit:
Правки v2

Внесены правки в соответсвии с полнотой комментариев к проектной работе

Commit:
правки v2.1

исправлены и оптимизированы ассерты

Commit:
правки v2.2

убран лишний класс для работы c GET// Orders

Commit:
правки v2.3

базовый урл вынесен в settings.py
в дата классах используем базовый урл + метод

AlexKitIt commented on data/settings.py:
```diff
@@ -0,0 +1,2 @@
+class BaseUrl:
+    base_url = "https://qa-scooter.praktikum-services.ru"
```
В этом классе можно было разместить и все урлы и все ендпоинты. Тогда и 3 модулей data не понадобилось бы, а был бы один
